onnxruntime_extensions/tools/add_pre_post_processing_to_model.py

```diff
@@ -162,6 +162,90 @@ def superresolution(model_file: Path, output_file: Path, output_format: str, onn
     onnx.save_model(new_model, str(output_file.resolve()))
 
 
+def yolo_detection(model_file: Path, output_file: Path, output_format: str,
+                   onnx_opset: int = 16, num_classes: int = 80):
+    """
+    SSD-like model and Faster-RCNN-like model are including NMS inside already, You can find it from onnx model zoo.
+
+    This function target for Yolo detection model. It support YOLOv3-yolov8 models theoretically.
+    You should assure this model has only one input, and the input shape is [1, 3, h, w].
+    The model has either one or more outputs. 
+        If the model has one output, the output shape is [1,num_boxes, coor+(obj)+cls] 
+            or [1, coor+(obj)+cls, num_boxes].
+        If the model has more than one outputs, you should assure the first output shape is 
+            [1, num_boxes, coor+(obj)+cls] or [1, coor+(obj)+cls, num_boxes].
+    Note: (obj) means it's optional.
+
+    :param model_file: The model file path.
+    :param output_file: The output file path.
+    """
+    model = onnx.load(str(model_file.resolve(strict=True)))
+    inputs = [create_named_value("image", onnx.TensorProto.UINT8, ["num_bytes"])]
+
+    model_input_shape = model.graph.input[0].type.tensor_type.shape
+    model_output_shape = model.graph.output[0].type.tensor_type.shape
+
+    # Should we use 640*640 as default input shape if no specific shape provided?
+    w_in = model_input_shape.dim[-1].dim_value if model_input_shape.dim[-1].HasField("dim_value") else 640
+    h_in = model_input_shape.dim[-2].dim_value if model_input_shape.dim[-1].HasField("dim_value") else 640
+
+    # We have no idea if we need to transpose the output or not.
+    # To run the model and get the real shape is the only way to know the concrete shape.
```

Comment:
The model may define the concrete shape (at least the exported yolov8 I have does) so we could check for dim_value's in the output shape.

Reply:
Would it be better to assume model has a symbolic shape? Will check output_shape if there is and perverse the backward shape-infer.